flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/ResultPartitionTest.java

```diff
@@ -100,6 +101,22 @@ public void testAddOnFinishedBlockingPartition() throws Exception {
 		testAddOnFinishedPartition(ResultPartitionType.BLOCKING);
 	}
 
+	@Test
+	public void testExternallyManagedBlockingPartitionConsumableMultipleTimes() throws IOException {
+		final ResultPartition partition = new ResultPartitionBuilder()
+			.setIsExternallyManaged(true)
+			.setResultPartitionType(ResultPartitionType.BLOCKING)
```

Comment:
I guess we do not need to set partition type and number of subpartitions, and the default would make sense.

Reply:
The partition being blocking is important for the test so I'd like to not omit it.

Reply:
Actually I thought the property of `isExternallyManaged` should control whether it could be consumed multiple times not related to `ResultPartitionType`. But in reality the pipelined type could not support this atm because of the `checkState(readView == null` while `PipelinedSubpartition#createReadView`. So we can keep the current way for setting the blocking type.